### PR TITLE
Add makefile command to simplify local development

### PR DESCRIPTION
**Pull request by sologgfun**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
this PR just simplify local image build by makefile.
it handle the different between Linux and Darwin.
after this ,  developer just need `make local-image` , Instead of some bash script by themselves.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -74,6 +74,12 @@ unit-test: vendor
 verify:
 	@./hack/verify-all.sh
 
+## local-image: Build local image for testing
+.PHONY: local-image
+local-image:
+	@./hack/local-image.sh
+
+
```
Comment (wzshiming) [inline Makefile:82]:
```suggestion
```

File: hack/local-image.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Get the name of the operating system using the uname command
+os_name=$(uname)
+set -e
+
+function main() {
```
Comment (wzshiming) [inline hack/local-image.sh:7]: It looks like there is a circular dependency between `Makefile` and `local-image.sh` which is not good, we'd better implement the logic of l`ocal-image.sh` directly in `Makefile`

**Review by wzshiming — CHANGES_REQUESTED** [review]
It looks like this PR accidentally added executable permissions to the `.go` file (`hack/verify-boilerplate.go`) please restore it

**Comment by wzshiming** [comment]
FYI, if you want to update a patch, just push to the same branch and you don't need to re-sent a PR

File: hack/local-image.sh (hunk 2 above)
Comment (sologgfun) [reply hack/local-image.sh:7]: ok, it sounds good ,I misunderstand `hack/verify-boilerplate.go` before,so I will patch this commit later, thanks.

**Comment by sologgfun** [comment]
@wzshiming 
Hello, I have completed the required modifications for this PR , do you have time to review it and see if there are any other issues?

File: Makefile [hunk 3]
```diff
@@ -74,6 +78,21 @@ unit-test: vendor
...
+	@echo "The operating system is Linux"
+	BINARY=kwok make build && IMAGE_PREFIX=local make image 
+else ifeq ($(OS),Darwin)
+	@echo "The operating system is Darwin" 
+	$(eval export BINARY_PLATFORMS = $(CONTAINER_PLATFORMS))
+	$(eval export IMAGE_PLATFORMS = $(CONTAINER_PLATFORMS))
+	BINARY=kwok make cross-build && IMAGE_PREFIX=local make cross-image
+else 
+@echo "The operating system is not Linux or Darwin"
+endif
```
Comment (wzshiming) [inline Makefile:94]:
```suggestion
ifeq ($(GOOS),linux)
	@make BINARY=kwok build && \
	  make image
else ifeq ($(GOOS),darwin)
	@make BINARY=kwok BINARY_PLATFORMS=linux/$(GOARCH) cross-build && \
	  make IMAGE_PLATFORMS=linux/$(GOARCH) cross-image
else 
	@echo "Unsupported OS: $(GOOS)"
endif
```

File: Makefile [hunk 4]
```diff
@@ -38,6 +38,10 @@ BINARY_NAME ?=
 
 STAGING ?= false
 
+OS = $(shell uname)
+
+CONTAINER_PLATFORMS = linux/$(shell go env GOARCH)
```
Comment (wzshiming) [inline Makefile:43]:
```suggestion
GOOS ?= $(shell go env GOOS)

GOARCH ?= $(shell go env GOARCH)
```

File: docs/contributing/development.md [hunk 5]
```diff
@@ -14,17 +14,8 @@ On a successful build, the binaries will be located in `./bin/$(go env GOOS)/$(g
...
-IMAGE_PREFIX=local make image
-```
-
-On Darwin:
-``` bash
-CONTAINER_PLATFORMS=linux/$(go env GOARCH)
-BINARY_PLATFORMS=${CONTAINER_PLATFORMS} BINARY=kwok make cross-build
-IMAGE_PLATFORMS=${CONTAINER_PLATFORMS} IMAGE_PREFIX=local make cross-image
+```bash
+make local-image
```
Comment (wzshiming) [inline docs/contributing/development.md:18]:
```suggestion
IMAGE_PREFIX=local make build-image
```

File: Makefile [hunk 6]
```diff
@@ -74,6 +78,21 @@ unit-test: vendor
 verify:
 	@./hack/verify-all.sh
 
+## local-image: Build local image for testing
+.PHONY: local-image
+local-image:
```
Comment (wzshiming) [inline Makefile:83]:
```suggestion
## build-image: Build binary and image 
.PHONY: build-image
build-image:
```

**Review by wzshiming — CHANGES_REQUESTED** [review]

**Comment by sologgfun** [comment]
All commits have been squash

**Pull request merged**